distribution/src/bin/elasticsearch

```diff
@@ -45,6 +49,14 @@ then
   fi
 fi
 
+if [[ $ENROLL_TO_CLUSTER = true ]]; then
```

Comment:
The idea here is that this part runs only when an explicit flag is passed by the user. We will attempt to do our auto-configuration of TLS based on the enrollment process and only if everything went well will the node start with that configuration. If anything fails, we roll-back the configuration and fail with an error message.